### PR TITLE
Enable aave collateral adaptor

### PR DESCRIPTION
**Pull request by cbrit**

### CODE REVIEW

File: proto/adaptors/aave/aave_v2_enable_asset_as_collateral_adaptor.proto [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import "adaptors/base.proto";
+
+// Represents call data for the Aave AToken adaptor, used to manage lending positions on Aave
+message AaveV2EnableAssetAsCollateralAdaptorV1 {
+    oneof function {
+        /***** BASE ADAPTOR FUNCTIONS *****/
+
+        // Represents function `revokeApproval(ERC20 asset, address spender)`
+        RevokeApproval revoke_approval = 1;
```
Comment (EricBolten) [inline proto/adaptors/aave/aave_v2_enable_asset_as_collateral_adaptor.proto:18]: Is this actually necessary for this adapter?

File: proto/cellar_v2.proto [hunk 2]
```diff
@@ -373,5 +374,7 @@ message AdaptorCall {
         CellarAdaptorV1Calls cellar_v1_calls = 15;
         // Represents function calls to the UniswapV3Adaptor V2
         UniswapV3AdaptorV2Calls uniswap_v3_v2_calls = 16;
+        // Represents function calls to the AaveV2EnableAssetAsCollatorAdaptor V1
+        AaveV2EnableAssetAsCollateralAdaptorV1Calls aave_v2_enable_asset_as_collateral_v1_calls = 17;
```
Comment (EricBolten) [inline proto/cellar_v2.proto:389]: Looks like we're adding this to the 2.0 proto but have not done 2.2.

File: steward/src/cellars/adaptors/aave_v2.rs [hunk 3]
```diff
@@ -309,3 +311,41 @@ pub(crate) fn aave_debt_token_adaptor_v2_calls(
 
     Ok(calls)
 }
+
+pub(crate) fn aave_v2_enable_asset_as_collateral_adaptor_v1_calls(
```
Comment (EricBolten) [inline steward/src/cellars/adaptors/aave_v2.rs]: Like the Uniswap case, this is a separate adaptor so we may want a separate file.

**Review by EricBolten — COMMENTED** [review]
A few comments.

File: proto/adaptors/aave/aave_v2_enable_asset_as_collateral_adaptor.proto (hunk 1 above)
Comment (cbrit) [reply proto/adaptors/aave/aave_v2_enable_asset_as_collateral_adaptor.proto:18]: It's a base function for all protos that derive from BaseAdaptor.sol so I do it by default

File: proto/cellar_v2.proto (hunk 2 above)
Comment (cbrit) [reply proto/cellar_v2.proto:389]: AdaptorCall is a top level type in the proto, this applies to both. It's split out in the handler files.

Comment (EricBolten) [reply proto/cellar_v2.proto:389]: Ah, I see that now, 2.2 is split for some files but not here, roger that.

Comment (EricBolten) [reply proto/cellar_v2.proto:389]: Though the changes you've made support this adaptor for 2.0 as well, wouldn't we need to add it to the Cellar V2.0 section of this file?

Comment (EricBolten) [reply proto/cellar_v2.proto:389]: Never mind, I see it now.

**Review by EricBolten — APPROVED** [review]
LGTM

**Pull request merged**